[2.9.0] Feature Flag - external-rules (#1361)
rancher/rancher-docs: versioned_docs/version-2.9/getting-started/installation-and-upgrade/installation-references/feature-flags.md

The following is a list of feature flags available in Rancher. If you've upgraded from a previous Rancher version, you may see additional flags in the Rancher UI, such as `proxy` or `dashboard` (both [discontinued](/versioned_docs/version-2.5/reference-guides/installation-references/feature-flags.md)):

- `continuous-delivery`: Allows Fleet GitOps to be disabled separately from Fleet. See [Continuous Delivery.](../../../how-to-guides/advanced-user-guides/enable-experimental-features/continuous-delivery.md) for more information.

> - `external-rules`: This flag affected [external `RoleTemplate` behavior](../../../how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/manage-role-based-access-control-rbac/cluster-and-project-roles.md#external-roletemplate-behavior) and is removed in v2.9.0 and later Rancher versions as the flag always remains on.

Reviewer: ```suggestion
`external-rules`: This flag affected [external `RoleTemplate` behavior](../../../how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/manage-role-based-access-control-rbac/cluster-and-project-roles.md#external-roletemplate-behavior). It is removed in Rancher v2.9.0 and later as the behavior is enabled by default.
```

Author: Applied suggestion, thanks Marty!

> - `external-rules`: This flag is removed in this current and later Rancher versions as the flag always remains on. With its default enablement, external `RoleTemplates` can be created only if the backing `ClusterRole` exists in the local cluster or the `ExternalRules` is set. Note that `escalate` permissions on `RoleTemplates` are required to create external `RoleTemplates` with `ExternalRules`. For context, the backing `ClusterRole` holds cluster rules and privileges, and shares the same `metadata.name` used in the `RoleTemplate` in your respective cluster referenced by the `ClusterRoleTemplateBinding/ProjectRoleTemplateBinding`.

Reviewer: 1. If the flag isn't accessible to v2.9 users, I don't think we should add details about it to the v2.9 docs. 
2. But the information about how external `RoleTemplates` behave is always going to be relevant to v2.9.0, since the flag can no longer be disabled by the user. So maybe describe this behavior on a page related to cluster roles and role templates? https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/manage-role-based-access-control-rbac/cluster-and-project-roles

Author: The flag is no longer accessible to users so I ended up revising the "Feature Flag" page to point to a new section that describes the behavior change in the "Cluster and Project Roles" page mentioned, let me know what you think. Thanks Marty!

The following table shows the availability and default values for some feature flags in Rancher. Features marked "GA" are generally available:

| Feature Flag Name             | Default Value | Status       | Available As Of | Additional Information |
| ----------------------------- | ------------- | ------------ | --------------- | ---------------------- |
| `continuous-delivery` | `true` | GA | v2.6.0 | |
| `external-rules` | v2.7.14: `false`, v2.8.5: `true` | Removed | v2.7.14, v2.8.5 | This flag affected [external `RoleTemplate` behavior](../../../how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/manage-role-based-access-control-rbac/cluster-and-project-roles.md#external-roletemplate-behavior). It is removed in Rancher v2.9.0 and later as the behavior is enabled by default. |
| `fleet`  | `true` | Can no longer be disabled | v2.6.0 | |
| `fleet`  | `true` | GA | v2.5.0 | |
| `harvester` | `true` | Experimental | v2.6.1 | |
| `legacy` | `false` for new installs, `true` for upgrades | GA | v2.6.0 | |
| `rke1-custom-node-cleanup`| `true` | GA | v2.6.0 | |
| `rke2` | `true` | Experimental | v2.6.0 | |

> | `token-hashing` | `false` for new installs, `true` for upgrades | GA | v2.6.0 | |

Reviewer 2: ```suggestion
| `token-hashing` | `false` for new installs, `true` for upgrades | GA | v2.6.0 | |
| `ui-sql-cache` | `false` | Highly experimental | v2.9.0 |
```
